TestProject.BL/Services/UserService.cs

```diff
@@ -1,36 +1,49 @@
-﻿using TestProject.DAL.Models;
-using TestProject.DAL.Repositories;
+﻿using TestProject.DAL.Repositories;
 using System.Threading.Tasks;
 using TestProject.BL.Models;
+using TestProject.BL.Mappers;
+using TestProject.DAL.Models;
 
 namespace TestProject.BL.Services
 {
     public class UserService : IUserService
     {
-        private IUserRepository _repo;
+        private IUserRepository _userRepo;
 
         public UserService(IUserRepository repo)
         {
-            _repo = repo;
+            _userRepo = repo;
         }
-        public async Task<UserLoginModel> AddUserAsync(UserLoginModel userLoginModel)
+        public async Task<UserLoginModel> AddOrUpdateUserAsync(UserLoginModel userLoginModel)
         {
-            if (!UserExists(userLoginModel.Email))
+            if (_userRepo.UserExists(userLoginModel.Email))
             {
-                var user = new User
+                var user = _userRepo.GetUser(userLoginModel.Email);
+                var userChanged = CheckUserForChanges(userLoginModel, user);
+                if (userChanged)
                 {
-                    FirstName = userLoginModel.FirstName,
-                    LastName = userLoginModel.LastName,
-                    Email = userLoginModel.Email
-                };
-                await _repo.CreateAsync(user);
+                    var newUser = UserMapper.MapUserLoginModelToUser(userLoginModel);
+                    newUser.Id = user.Id;
+                    _userRepo.Update(newUser);
+                }
+            }
+            else
+            {
+                var user = UserMapper.MapUserLoginModelToUser(userLoginModel);
+                await _userRepo.CreateAsync(user);
             }
             return userLoginModel;
         }
 
-        public bool UserExists(string email)
+        public UserProfile GetUserProfile(string email)
+        {
+            var user = _userRepo.GetUser(email);
+            return UserMapper.MapUserToUserProfile(user);
+        }
+        
+        private bool CheckUserForChanges(UserLoginModel userLoginModel, User user)
```

Comment:
Name of methods that return `bool` should be a question that return value answers.
`HasCanged()` for example.